diff --git a/src/INTERP_KERNEL/MCIdType.hxx b/src/INTERP_KERNEL/MCIdType.hxx
index 91086b35a..0c9dd99c4 100644
--- a/src/INTERP_KERNEL/MCIdType.hxx
+++ b/src/INTERP_KERNEL/MCIdType.hxx
@@ -36,7 +36,7 @@ typedef std::int64_t mcIdType;
 
 #endif
 
-#ifdef WIN32
+#if defined(WIN32) || defined (__APPLE__)
 typedef long long mcPyPtrType;
 #else
 typedef long mcPyPtrType;
diff --git a/src/MEDCoupling_Swig/CMakeLists.txt b/src/MEDCoupling_Swig/CMakeLists.txt
index 0edd729a3..e2b5c54c1 100644
--- a/src/MEDCoupling_Swig/CMakeLists.txt
+++ b/src/MEDCoupling_Swig/CMakeLists.txt
@@ -100,7 +100,7 @@ if(MEDCOUPLING_BUILD_DOC)
        ${PROJECT_BINARY_DIR}/doc/MEDCoupling_doc.i swig_ready)
 endif()
 
-if(WIN32)
+IF(WIN32 OR APPLE)
   # Under Windows MEDCoupling conflicts wiht medcoupling from PyWrapping
   set(MEDCouling_target_name MEDCouplingCompat)
   set_property(SOURCE MEDCoupling.i PROPERTY SWIG_MODULE_NAME MEDCouplingCompat)
@@ -133,7 +133,7 @@ endif()
 
 swig_check_generation(${MEDCouling_target_name})
 set_source_files_properties(MEDCouplingRemapper.i PROPERTIES CPLUSPLUS ON)
-if(WIN32)
+IF(WIN32 OR APPLE)
   set_property(SOURCE MEDCouplingRemapper.i PROPERTY COMPILE_DEFINITIONS WIN32)
 endif()
 if("${PYTHON_VERSION_MAJOR}" STREQUAL "3")
@@ -184,7 +184,7 @@ install(TARGETS ${SWIG_MODULE_${MEDCouling_target_name}_REAL_NAME}
         DESTINATION ${MEDCOUPLING_INSTALL_PYTHON})
 
 set(PYFILES_TO_INSTALL ${CMAKE_CURRENT_BINARY_DIR}/MEDCouplingRemapper.py)
-if(WIN32)
+IF(WIN32 OR APPLE)
   set(PYFILES_TO_INSTALL ${PYFILES_TO_INSTALL}
                          ${CMAKE_CURRENT_BINARY_DIR}/MEDCouplingCompat.py)
 else()
diff --git a/src/MEDLoader/Swig/CMakeLists.txt b/src/MEDLoader/Swig/CMakeLists.txt
index cb44ed0a2..1ee15ce61 100644
--- a/src/MEDLoader/Swig/CMakeLists.txt
+++ b/src/MEDLoader/Swig/CMakeLists.txt
@@ -72,7 +72,7 @@ if(MEDCOUPLING_BUILD_DOC)
        ${PROJECT_BINARY_DIR}/doc/MEDLoader_doc.i swig_ready)
 endif()
 
-if(WIN32)
+IF(WIN32 OR APPLE)
   set_property(SOURCE MEDLoader.i PROPERTY COMPILE_DEFINITIONS WIN32)
 endif()
 
diff --git a/src/ParaMEDMEM_Swig/CMakeLists.txt b/src/ParaMEDMEM_Swig/CMakeLists.txt
index 6cad73b98..542a3f93c 100644
--- a/src/ParaMEDMEM_Swig/CMakeLists.txt
+++ b/src/ParaMEDMEM_Swig/CMakeLists.txt
@@ -60,6 +60,10 @@ set(SWIG_MODULE_ParaMEDMEM_EXTRA_DEPS
     ${medcoupling_HEADERS_HXX} ${medcoupling_HEADERS_TXX}
     ${interpkernel_HEADERS_HXX} ${interpkernel_HEADERS_TXX})
 
+IF(APPLE)
+  SET_PROPERTY(SOURCE ParaMEDMEM.i PROPERTY COMPILE_DEFINITIONS WIN32)
+ENDIF()
+
 if(${CMAKE_VERSION} VERSION_LESS "3.8.0")
   swig_add_module(ParaMEDMEM python ParaMEDMEM.i)
 else()
diff --git a/src/PyWrapping/CMakeLists.txt b/src/PyWrapping/CMakeLists.txt
index d61d9fbcc..86f183d9f 100644
--- a/src/PyWrapping/CMakeLists.txt
+++ b/src/PyWrapping/CMakeLists.txt
@@ -28,7 +28,7 @@ add_definitions(
   -DMEDCOUPLING_CONFIG_DT=\"${MEDCOUPLING_CONFIG_TIME}\")
 
 set_source_files_properties(medcoupling.i PROPERTIES CPLUSPLUS ON)
-if(WIN32)
+IF(WIN32 OR APPLE)
   set_property(SOURCE medcoupling.i PROPERTY COMPILE_DEFINITIONS WIN32)
 endif()
 if("${PYTHON_VERSION_MAJOR}" STREQUAL "3")
diff --git a/src/RENUMBER_Swig/CMakeLists.txt b/src/RENUMBER_Swig/CMakeLists.txt
index 246287599..4aff75219 100644
--- a/src/RENUMBER_Swig/CMakeLists.txt
+++ b/src/RENUMBER_Swig/CMakeLists.txt
@@ -57,7 +57,7 @@ include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/../INTERP_KERNEL/ExprEval
   ${CMAKE_CURRENT_SOURCE_DIR}/../INTERP_KERNEL/GaussPoints)
 
-if(WIN32)
+IF(WIN32 OR APPLE)
   set_property(SOURCE MEDRenumber.i PROPERTY COMPILE_DEFINITIONS WIN32)
 endif()
 if(${CMAKE_VERSION} VERSION_LESS "3.8.0")
